### PR TITLE
Web preview

### LINKED ISSUE #243: Web Preview
Add a preview of how style and layout settings will appear in Apple News.

### PR DESCRIPTION
**Pull request by bcampeau**
This enables a web preview that gives an approximate representation in HTML of what your Apple News settings will look like in Apple News.

Closes #243

### CODE REVIEW

File: admin/apple-actions/index/class-export.php [hunk 1]
```diff
@@ -85,7 +133,7 @@ public function fetch_exporter() {
 			preg_match( '/#(.*?)#/', $byline, $matches );
 			if ( ! empty( $matches[1] ) ) {
 				// Set the date using the custom format
-				$byline = str_replace( $matches[0], date( $matches[1], strtotime( $post->post_date ) ), $byline );
+				$byline = str_replace( $matches[0], date( $matches[1], strtotime( $date ) ), $byline );
```
Comment (mslinnea) [inline admin/apple-actions/index/class-export.php]: Looks like $date could still be empty here.

File: admin/partials/page_options.php [hunk 2]
```diff
@@ -1,8 +1,9 @@
-<div class="wrap">
-	<form method="post" action="">
+<div class="wrap apple-news-settings">
+	<form method="post" action="" id="apple-news-settings-form">
 		<?php wp_nonce_field( 'apple_news_options', 'apple_news_options' ); ?>
 		<input type="hidden" name="action" value="apple_news_options" />
 		<?php foreach ( $sections as $section ): ?>
+		<?php $section->before_section() ?>
```
Comment (mslinnea) [inline admin/partials/page_options.php]: this block should be indented

File: admin/settings/class-admin-apple-settings-section-formatting.php [hunk 3]
```diff
@@ -237,6 +237,71 @@ public function get_section_info() {
...
+						echo wp_kses( $$component, self::$allowed_html );
+					}
+				?>
+				<div class="apple-news-component">
+				<p><span class="apple-news-dropcap">L</span>orem ipsum dolor sit amet, consectetur adipiscing elit. Mauris sagittis, <a href="#">augue vitae iaculis euismod</a>, libero nulla pellentesque quam, non venenatis massa odio id dolor.</p>
+				<div class="apple-news-pull-quote">Lorem ipsum dolor sit amet.</div>
+				<p>Praesent eget odio vel sapien scelerisque euismod. Phasellus eros sapien, rutrum ac nibh nec, tristique commodo neque.</p>
+				<h2>Quisque efficitur</h2>
+				<p>Quisque efficitur sit amet ex et venenatis. Morbi nisi nisi, ornare id iaculis eget, pulvinar ac dolor.</p>
+				<p>In eu la	cus porttitor, pellentesque diam et, tristique elit. Mauris justo odio, efficitur sit amet aliquet id, aliquam placerat turpis.</p>
```
Comment (mslinnea) [inline admin/settings/class-admin-apple-settings-section-formatting.php]: Is this placeholder text supposed to be here?

**Comment by mslinnea** [comment]
🐴

File: admin/apple-actions/index/class-export.php (hunk 1 above)
Comment (bcampeau) [inline admin/apple-actions/index/class-export.php]: OK, good point.

File: admin/partials/page_options.php (hunk 2 above)
Comment (bcampeau) [inline admin/partials/page_options.php]: OK

File: admin/settings/class-admin-apple-settings-section-formatting.php (hunk 3 above)
Comment (bcampeau) [inline admin/settings/class-admin-apple-settings-section-formatting.php]: It's dummy content for a preview. Intentional.

**Pull request merged**